.github/workflows/ci.yml

```diff
@@ -28,6 +28,40 @@ jobs:
       - name: Check formatting
         run: npm run check-formatting
 
+      - name: Compare synced directories between PR and base
+        # if the label exists, any changes are fine, so no need to check
+        if: >-
+          ! contains(github.event.pull_request.labels.*.name , 'automation:sync-docs')
+        run: |
+          autogened=(
+             docs/docs
+             docs/reference/help-all.json
+             versioned_docs/*/docs
+             versioned_docs/*/reference/help-all.json
+          )
+
+          git fetch --depth=1 origin ${{ github.event.pull_request.base.ref }}
+
+          if ! git diff --quiet FETCH_HEAD HEAD -- "${autogened[@]}"; then
+            echo "::group::Full diff"
+            git diff FETCH_HEAD HEAD -- "${autogened[@]}"
+            echo "::endgroup::"
+
+            cat <<EOF
+          Error: found changes to files that are synced from <https://github.com/pantsbuild/pants>:
+
+          $(git diff --name-only FETCH_HEAD HEAD -- "${autogened[@]}")
+
+          Those files should be changed in <https://github.com/pantsbuild/pants>, and are
+          automatically synced to this repository after each release.
+
+          (If this change is intentional and you're sure it's appropriate, add the
+          'automation:sync-docs' label.)
```

Comment:
Can we add a new label for the human override so the `automation:sync-docs` label remains for automation?

Something like `bypass-sync-check`

Reply:
I would lean towards keeping a simpler set of labels, since I wouldn't expect humans to need to do this very often (once we get through the initial setup phase). If we find people are needing to use it a lot, we can always change then.